pageserver/src/layered_repository.rs

```diff
@@ -1381,25 +1383,46 @@ impl LayeredTimeline {
             page_img: None,
         };
 
-        if let Some(_cont_lsn) = layer.get_page_reconstruct_data(blknum, lsn, &mut data)? {
-            // The layers are currently fully self-contained, so we should have found all
-            // the data we need to reconstruct the page in the layer.
-            if data.records.is_empty() {
-                // no records, and no base image. This can happen if PostgreSQL extends a relation
-                // but never writes the page.
-                //
-                // Would be nice to detect that situation better.
-                warn!("Page {} blk {} at {} not found", seg.rel, blknum, lsn);
-                return Ok(ZERO_PAGE.clone());
+        // Holds an Arc reference 'curr_layer' when iterating in the loop below.
+        let mut last_layer: Arc<dyn Layer>;
+
+        // Call the layer's get_page_reconstruct_data function to get the base image
+        // and WAL records needed to materialize the page. If it returns 'Continue',
+        // call it again on the predecessor layer until we have all the required data.
+        let mut curr_layer = layer;
+        let mut curr_lsn = lsn;
```

Comment:
It took me a few reads to understand what was going on here.
Would it make sense to rename 
`last_layer` -> `layer_arc`
`curr_layer` -> `layer_ref`

Reply:
Done